app/src/test/java/io/gnosis/safe/ui/settings/safe/SafeSettingsEditNameViewModelTest.kt

```diff
@@ -0,0 +1,62 @@
+package io.gnosis.safe.ui.settings.safe
+
+import io.gnosis.data.models.Safe
+import io.gnosis.data.repositories.SafeRepository
+import io.gnosis.safe.MainCoroutineScopeRule
+import io.gnosis.safe.TestLifecycleRule
+import io.gnosis.safe.TestLiveDataObserver
+import io.gnosis.safe.appDispatchers
+import io.gnosis.safe.ui.base.BaseStateViewModel
+import io.mockk.*
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.Rule
+import org.junit.Test
+import pm.gnosis.model.Solidity
+import java.math.BigInteger
+
+class SafeSettingsEditNameViewModelTest {
+
+    @get:Rule
+    val coroutineScope = MainCoroutineScopeRule()
+
+    @get:Rule
+    val instantExecutorRule = TestLifecycleRule()
+
+    private val safeRepository = mockk<SafeRepository>(relaxed = true)
+    private lateinit var viewModel: SafeSettingsEditNameViewModel
+
+    @Test
+    fun `saveLocalName (name) - should change local name for safe`() = runBlockingTest {
+
```

Comment:
This only tests the happy case. Would love to see test for some edge cases (e.g. no safe or error when interacting with the repo). But not super important right now.